pkg/kubectl/cmd/testing/fake.go

```diff
@@ -244,9 +249,45 @@ type TestFactory struct {
 	OpenAPISchemaFunc                func() (openapi.Resources, error)
 }
 
+func NewDefaultFakeClientConfigLoadingRules() (*clientcmd.ClientConfigLoadingRules, error) {
+	tempConfigFile, err := ioutil.TempFile("", "cmdtests_temp")
```

Comment:
where does this get cleaned up? 
shouldn't it explicitly be in `tempDir` instead of relying on them to match?

Reply:
https://golang.org/pkg/io/ioutil/#TempFile
> If dir is the empty string, TempFile uses the default directory for temporary files (see os.TempDir).

the `tempDir` and `filepath.Join` bit below may not be needed at all?

Reply:
yes, the tempDir + join smells fishy, but if we do need them, then we should actually specify the dir here, otherwise we have some weird implementation-detail coupling below.

Reply:
Thanks, cleaned this up a bit.
As for the file cleanup, I added a `Cleanup` method to the fake factory that removes the temp file.
Now I just need to call it from all callers of NewTestFactory()

Reply:
Great!